flink-connectors/flink-connector-files/src/main/java/org/apache/flink/connector/file/src/assigners/LocalityAwareSplitAssigner.java

```diff
@@ -69,9 +70,10 @@ public class LocalityAwareSplitAssigner implements FileSplitAssigner {
 
     // --------------------------------------------------------------------------------------------
 
-    public LocalityAwareSplitAssigner(Collection<FileSourceSplit> splits) {
-        for (FileSourceSplit split : splits) {
-            this.unassigned.add(new SplitWithInfo(split));
+    public LocalityAwareSplitAssigner(Iterator<FileSourceSplit> splits) {
+        while (splits.hasNext()) {
```

Comment:
Actaully, it'll still cause OOM for in here we will still maintain **all splits** in memory.